std/algorithm/searching.d

```diff
@@ -1457,35 +1447,51 @@ private auto extremum(alias selector = "a < b", Range,
 
 // find
 /**
-Finds an individual element in an input range. Elements of $(D
-haystack) are compared with `needle` by using predicate $(D
-pred). Performs $(BIGOH walkLength(haystack)) evaluations of $(D
-pred).
+Finds an individual element in an $(REF_ALTTEXT input range, isInputRange, std,range,primitives).
+Elements of `haystack` are compared with `needle` by using predicate
+`pred` with `pred(haystack.front, needle)`.
+`find` performs $(BIGOH walkLength(haystack)) evaluations of `pred`.
+
+To _find the last occurrence of `needle` in `haystack` of a
+$(REF_ALTTEXT bidirectional, isBidirectionalRange, std,range,primitives) `haystack`,
+call `find(retro(haystack), needle)`. See $(REF retro, std,range).
+
+If not `needle` is provided, `pred(haystack.front)` will be evaluated on each
+element of the input range.
 
-To _find the last occurrence of `needle` in `haystack`, call $(D
-find(retro(haystack), needle)). See $(REF retro, std,range).
+If `input` is a $(REF_ALTTEXT forward range, isForwardRange, std,range,primitives),
+`needle` can be a $(REF_ALTTEXT forward range, isForwardRange, std,range,primitives) too.
+In this case `startsWith!pred(haystack, needle)` is evaluated on each evaluation.
+
+Note:
+    `find` behaves similar to `dropWhile` in other languages.
+
+Complexity:
+    `find` performs $(BIGOH walkLength(haystack)) evaluations of `pred`.
+    There are specializations that improve performance by taking
+    advantage of $(REF_ALTTEXT bidirectional range, isBidirectionalRange, std,range,primitives)
+    or random access in the given ranges (where possible), depending on the statistics
+    of the two ranges' content.
```

Comment:
...advantage of bidirectional or random access ranges, depending...

Note: as a reader, I'm not sure I'd understand what these statistics are.. by statistics do you mea type qualifiers i.e. const/immutable/etc?

Reply:
That was moved 1:1 from the existing overload. It's hard for me to understand as well and I guess whoever wrote this wanted to refer to taking advantage of a range e.g. being random access.

I simply removed this confusing bit.